### ISSUE TITLE
Refactoring of custom schema request handler 

### ISSUE THREAD

**Issue by tishion** [tishion/QCefView#5]
refactor the custom schema request handler

**Comment by Connlaio** [comment]
tishion:  I cloned your code from the github and build it with  Qt5.3.1 + MSVC2013,the cef version is 2556,when I tried to run the project Test,it will crashed in the error is "Debug Assertion Failed! Expression: _pFirstBlock == pHead", and I tried to debug it ,found the error is in the CCefSetting construct fuction. Did you fixed it ?

**Comment by Connlaio** [comment]
tishion:  I cloned your code from the github and build it with  Qt5.3.1 + MSVC2013,the cef version is 2556,when I tried to run the project Test,it will crashed in the error is "Debug Assertion Failed! Expression: _pFirstBlock == pHead", and I tried to debug it ,found the error is in the CCefSetting construct fuction. Did you fixed it ?

**Comment by Connlaio** [comment]
tishion:  I cloned your code from the github and build it with  Qt5.3.1 + MSVC2013,the cef version is 2556,when I tried to run the project Test,it will crashed in the error is "Debug Assertion Failed! Expression: _pFirstBlock == pHead", and I tried to debug it ,found the error is in the CCefSetting construct fuction. Did you fixed it ?

**Comment by Connlaio** [comment]
tishion:  I cloned your code from the github and build it with  Qt5.3.1 + MSVC2013,the cef version is 2556,when I tried to run the project Test,it will crashed in the error is "Debug Assertion Failed! Expression: _pFirstBlock == pHead", and I tried to debug it ,found the error is in the CCefSetting construct fuction. Did you fixed it ?

**Comment by halsten** [comment]
Yeah, same for me. It Crashes. After spending lengthy time to have it compile and run, I was upset that I didn't get to see anything. Any chance you would fix it or explain whats going on?

**Comment by tishion** [comment]
Hi @halsten I believe it is caused by the runtime configuration of you compiling environment. It seems I need to reform the build tools. Please give me some time.

**Comment by halsten** [comment]
@tishion Thanks a lot for the reply! Please, take your time. I would like to be able to run it or at least just get to see you bundling CEF with Qt.

**Comment by lgxZJ** [comment]
@tishion  does QCefView supports WinXp?

**Comment by tishion** [comment]
@lgxZJ  it does.  but you need to choose to the CEF version which supports XP.

**Comment by lgxZJ** [comment]
@tishion  Great! But where can i find the XP version of CEF ? I  only see minimal distribution and standard distribution under this website http://opensource.spotify.com/cefbuilds/index.html

**Comment by tishion** [comment]
I remember the one in the repo does support xp.

**Comment by lgxZJ** [comment]
@tishion Thanks a lot, but i don't see any "XP" keyward in the automated builds. Is the website wrong? Can you give me more tips?

**Comment by tishion** [comment]
@lgxZJ  It doesn't matter, you can choose any version of CEF.
What you need to do is:
1. Build the libcef_dll wrapper first, this is a static library, when you build it just choose the toolset which supports XP. 
2. Use the libcef_dll wrapper you build to build the QCefView 

When you choose to support XP you need to use the correct Qt library which supports XP too.

Just check out the repo and you will find the libcef_dll wrapper projects in in the solution.

**Comment by lgxZJ** [comment]
@tishion It's very kind of you to give me such suggestions. The problem i encountered is that the "libcef.dll" in the CEF version i downloaded depends on a Vista system function, so i cannot use CEF in XP, Qt and libcef_dll_wrapper.lib works fine. That's why i am eager to known the version which support XP.

Is is possible to build the libcef.dll from source easily?

**Comment by lgxZJ** [comment]
@tishion  Oh, i find it in this place  
https://bitbucket.org/chromiumembedded/cef/wiki/BranchesAndBuilding#markdown-header-background.

Thanks a lot !!

**Comment by tishion** [comment]
@lgxZJ   you know from this page http://opensource.spotify.com/cefbuilds/index.html
you can download the CEF binary distributed version (the dllls are built with specified toolset, maybe they are not support XP.) 
If you really want to build CEF by yourself,  you need to checkout the CEF source and chrome source(quite large).

you can try to find cef_binary_3.2526.1373 this version supports XP, I didn't find it.

you can try this version:
http://www.teahoo.net/downfile/cef_binary_3.2526.1366.g8617e7c_windows32.rar

**Comment by lgxZJ** [comment]
@tishion    I find the version i need, but your reply is also helpful. Thanks, gentle man(^_^)!

**Comment by tishion** [comment]
@lgxZJ  good luck!
